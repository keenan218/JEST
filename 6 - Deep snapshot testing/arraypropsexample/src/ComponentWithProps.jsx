import React from "react";
import PropTypes from "prop-types";

const ComponentWithProps = (props) => {
  const { arrayProp, headerText } = props;

  const nextNumberDisplay = arrayProp.map((seqNum) => (
    <p key={seqNum} className="nd">
      Number is : {seqNum}; Next next is {(seqNum += 1)}
    </p>
  ));

  return (
    <div>
      <h1>{headerText}</h1>
      {nextNumberDisplay}
    </div>
  );
};
ComponentWithProps.defaultProps={
  headerText: `A default setting`
}
ComponentWithProps.propTypes = {
  headerText: PropTypes.string,
  arrayProp: PropTypes.array,
};

export default ComponentWithProps;
