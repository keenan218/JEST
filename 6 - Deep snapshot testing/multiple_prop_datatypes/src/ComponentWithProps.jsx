import React from 'react'; 
import PropTypes from 'prop-types';

const ComponentWithProps = props =>{
    const {headerText} = props; 
    return(<h1>{headerText}</h1>);
}
ComponentWithProps.defaultProps = {
    headerText: `Text supplied by the default value of headerText`,
};

ComponentWithProps.propTypes={
    headerText: PropTypes.string
}
export default ComponentWithProps