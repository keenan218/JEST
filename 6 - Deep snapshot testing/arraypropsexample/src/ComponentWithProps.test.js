import React from 'react';
import renderer from 'react-test-renderer';

import ComponentWithProps from './ComponentWithProps';

describe("ComponentWithProps render tests", () => {
    let componentToTest;

    let props = {
        headerText: `A header`,
        arrayProp:[1,2,3],
    };

    beforeEach(()=>{
        const testInstance = renderer.create(<ComponentWithProps {...props}/>);
        componentToTest = testInstance.root; 
    });

    it(`should render 'a header' in a h1 element`, () => {
        const h1render = componentToTest.findByType("h1");
        expect(h1render.children).toEqual([props.headerText]);
    })

    it(`should render the correct values when rendering the nextNumberDisplay`, () => {
        const nextNumberParas = componentToTest.findAllByProps({ className: "nd" });

        for(let i=0; i<props.arrayProp.length; i++){
            expect(nextNumberParas[i].children).toContain(props.arrayProp[i].toString(), (props.arrayProp[i] + 1).toString());
            expect(nextNumberParas[i].children).toContain(props.arrayProp[i].toString(), (props.arrayProp[i] + 1).toString());
        }
        expect(props.arrayProp).toMatchSnapshot();
    });
})