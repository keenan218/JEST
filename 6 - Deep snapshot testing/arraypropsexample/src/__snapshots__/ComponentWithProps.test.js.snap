// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentWithProps render tests should render the correct values when rendering the nextNumberDisplay 1`] = `
Array [
  1,
  2,
  3,
]
`;
